server/src/FeatureFinder.ts

```diff
@@ -50,66 +32,69 @@ export class FeatureFinder {
-   * @returns {Command} or `undefined` if nothing is found
-   */
-  async findDbtRpcCommand(dbtProfileType?: string): Promise<Command | undefined> {
+  async getAvailableDbt(): Promise<[DbtVersionInfo?, DbtVersionInfo?, DbtVersionInfo?, DbtVersionInfo?]> {
     const settledResults = await Promise.allSettled([
-      this.findDbtRpcPythonVersion(dbtProfileType),
-      this.findDbtPythonVersion(dbtProfileType),
-      this.findDbtRpcGlobalVersion(dbtProfileType),
+      this.findDbtRpcPythonInfo(),
+      this.findDbtPythonInfo(),
+      this.findDbtRpcGlobalInfo(),
+      this.findDbtGlobalInfo(),
     ]);
-
-    const [dbtRpcPythonVersion, dbtPythonVersion, dbtRpcGlobalVersion] = settledResults.map(v => {
+    const [dbtRpcPythonVersion, dbtPythonVersion, dbtRpcGlobalVersion, dbtGlobalVersion] = settledResults.map(v => {
       return v.status === 'fulfilled' ? v.value : undefined;
     });
 
-    let versions = '';
-    versions += dbtRpcGlobalVersion ? `dbtRpcGlobalVersion = ${getStringVersion(dbtRpcGlobalVersion.installedVersion)} ` : '';
-    versions += dbtPythonVersion ? `dbtPythonVersion = ${getStringVersion(dbtPythonVersion.installedVersion)} ` : '';
-    versions += dbtRpcPythonVersion ? `dbtRpcPythonVersion = ${getStringVersion(dbtRpcPythonVersion.installedVersion)}` : '';
+    console.log(
+      this.getLogString('dbtRpcGlobalVersion', dbtRpcGlobalVersion) +
+        this.getLogString('dbtPythonVersion', dbtPythonVersion) +
+        this.getLogString('dbtRpcPythonVersion', dbtRpcPythonVersion) +
+        this.getLogString('dbtGlobalVersion', dbtGlobalVersion),
+    );
+
+    return [dbtRpcPythonVersion, dbtPythonVersion, dbtRpcGlobalVersion, dbtGlobalVersion];
+  }
 
-    console.log(versions);
+  getLogString(name: string, dbtVersionInfo?: DbtVersionInfo): string {
+    return dbtVersionInfo ? `${name} = ${getStringVersion(dbtVersionInfo.installedVersion)} ` : '';
+  }
+
+  /** Tries to find a suitable command to start the server first in the current Python environment and then in the global scope.
+   * Installs dbt-rpc for dbt version > 1.0.0.
+   * @returns {Command} or `undefined` if nothing is found
+   */
+  async findDbtRpcCommand(dbtProfileType?: string): Promise<Command | undefined> {
+    const [dbtRpcPythonVersion, dbtPythonVersion, dbtRpcGlobalVersion] = await this.availableCommandsPromise;
 
-    if (dbtRpcPythonVersion?.installedVersion && dbtRpcPythonVersion.installedAdapter) {
+    if (dbtRpcPythonVersion?.installedVersion && dbtRpcPythonVersion.installedAdapters.some(a => a.name === dbtProfileType)) {
       this.versionInfo = dbtRpcPythonVersion;
-      return new DbtRpcCommand(FeatureFinder.DBT_RPC_PARAMS, this.pythonInfo?.path);
+      return this.dbtCommandFactory.getDbtRpcRun();
     }
     if (dbtPythonVersion?.installedVersion) {
       this.versionInfo = dbtPythonVersion;
       return dbtPythonVersion.installedVersion.major >= 1
         ? this.installAndFindCommandForV1(dbtProfileType)
```

Comment:
What if CLI mode is selected? It seems that we should not install Dbt RPC in this case.

Reply:
I skipped installation for `win32` (see `getFullDbtInstallationPackages` method), for other platform we still can use it.